native/src/consensus/mod.rs

```diff
@@ -0,0 +1,190 @@
+
+    for addr in AddrStream::new(GeoStream::new(Some(query_points)), context.clone(), args.error_query_points) {
+        for source in &sources {
+            // pull the addresses matching this address number within 1 km
+            let rows = conn.query(query, &[ &addr.number, &addr.geom[0], &addr.geom[1], &source ]).unwrap();
+
+            // populate potential_matches with db response
+            let mut potential_matches: Vec<Address> = Vec::with_capacity(rows.len());
+            for row in rows.iter() {
+                let paddr: serde_json::Value = row.get(1);
+                let paddr = Address::from_value(paddr).unwrap();
+                potential_matches.push(paddr);
+            }
+
+            // find a match using linker
+            match compare(&addr, &mut potential_matches) {
+                Some(link_id) => {
+                    let mut pmatches: Vec<Address> = potential_matches.into_iter().filter(|potential| {
+                        link_id == potential.id.unwrap()
+                    }).collect();
+
+                    match pmatches.len() {
+                        0 => continue,
+                        1 => {
+                            let paddr = pmatches.pop();
+                            let coords = match paddr {
+                                Some(point) => {
+                                    Some((point.geom[0], point.geom[1]))
+                                },
+                                None => None
+                            };
+                            // update source_map with current match for source
+                            source_map.entry(source.to_string()).and_modify(|e| *e = coords);
+                        },
+                        _ => panic!("Duplicate IDs are not allowed in input data")
+                    }
+                },
+                None => ()
+            };
+        }
+
+        // update agreement with current set of matched points
+        agreement.process_points(&source_map);
+    }
+
+    Ok(neon_serde::to_value(&mut cx, &agreement)?)
+}
+
+///
+/// Compare a given address against a list of proximal addresses
+///
+/// The function will return None if the address does not exist in the
+/// proximal set and should be considered a new address
+///
+/// The function will return Some(i64) if the address matches an existing address
+///
+pub fn compare(addr: &Address, potentials: &mut Vec<Address>) -> Option<i64> {
+    // No nearby addresses with this number
+    if potentials.len() == 0 {
+        return None;
+    }
+    let addr_link = linker::Link::new(0, &addr.names);
+    let potential_links: Vec<linker::Link> = potentials.iter().map(|potential| {
+        linker::Link::new(potential.id.unwrap(), &potential.names)
+    }).collect();
+
+    match linker::linker(addr_link, potential_links, true) {
+        Some(link) => Some(link.id),
+        None => None
+    }
```

Comment:
[nit] Another case where `Option.map` might be a better fit
```suggestion
    linker::linker(addr_link, potential_links, true).map(|link| link.id);
````